src/libs/MoneyRequestReportUtils.ts

```diff
@@ -39,4 +40,26 @@ function getThreadReportIDsForTransactions(reportActions: ReportAction[], transa
         .filter((reportID): reportID is string => !!reportID);
 }
 
-export {isActionVisibleOnMoneyRequestReport, getThreadReportIDsForTransactions};
+/**
+ * Filters all available transactions and returns the ones that belong to a specific report (by `reportID`).
+ * It is used as an onyx selector, to make sure that report related views do not process all transactions in onyx.
+ */
+function selectAllTransactionsForReport(transactions: OnyxCollection<Transaction>, reportID: string | undefined, reportActions: ReportAction[]) {
+    if (!reportID) {
+        return [];
+    }
+
+    return Object.values(transactions ?? {}).filter((transaction): transaction is Transaction => {
+        if (!transaction) {
+            return false;
+        }
+        const action = getIOUActionForTransactionID(reportActions, transaction.transactionID);
+        return transaction.reportID === reportID && !isDeletedParentAction(action);
+    });
+}
+
+function isSingleTransactionReport(transactions: Transaction[]) {
+    return transactions.length === 1;
+}
```

Comment:
I would rename it to something like
```suggestion
function isSingleItemTransactionList(transactions: Transaction[]) {
    return transactions.length === 1;
}
```
because right now the function name sounds like the parameter is actually report and it is not.

Also, if we want to have a function like this, we may want to modify other code fragments that compare transaction lengths to be equal to 1 using this method. For example, here:

https://github.com/Expensify/App/blob/d8cc586f5bb3ac748878aa7a0061eb2b87f49480/src/components/MoneyReportHeader.tsx#L207-L223

Reply:
yeah I understand the possible confusion, but I explicitly want this function to be used in context of reports. I don't want other devs to use to test random lists of transactions.
This is responsible for exactly 1 specific case - which is to decide whether a report is of the special "single transaction" case.

In your example is the code that you link specific to "single transaction/expense report"? If not - then that is exactly my point, I don't want this function to be used.
If yes then we should use it 👍